### ISSUE TITLE
Performance test failing on 100-node cluster with ResourceQuotas enabled

### ISSUE THREAD

**Issue by shyamjvs** [kubernetes/kubernetes#60988]
Based on https://github.com/kubernetes/kubernetes/issues/60589#issuecomment-371876879

After enabling quotas (specifically pods per namespace) by default in our tests, we're continuously seeing such failures:

```
Container kube-controller-manager-e2e-big-master/kube-controller-manager is using 0.531144723/0.5 CPU
not to have occurred
```

This may be a serious issue with scalability of quotas and needs some digging in.

/assign @gmarek 
(who enabled it.. feel free to redirect this as apt)

cc @kubernetes/sig-scalability-bugs

**Comment by shyamjvs** [comment]
I'm not marking this as a release-blocker as it's failures we started seeing due to testing something new (that we weren't before).
However, feel free to override me if you think otherwise.

**Comment by shyamjvs** [comment]
For now I'll send a revert disabling it. However, you can easily experiment against 100-node cluster with the presubmit job I newly added. See https://github.com/kubernetes/kubernetes/issues/56032#issuecomment-371229089 for details.

**Comment by shyamjvs** [comment]
One thing to note here is that, even before this change we were seeing flakes on this kind. E.g:

- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/11453
- https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/ci-kubernetes-e2e-gci-gce-scalability/11439

But after this change, we're seeing those continuously. My feeling is that this worsened things on top of something bad that happened already earlier.

**Comment by tpepper** [comment]
@kubernetes/sig-scalability-bugs do you see this as a v1.10 milestone issue?

**Comment by shyamjvs** [comment]
Not really.. for the reason I mentioned in https://github.com/kubernetes/kubernetes/issues/60988#issuecomment-371878759

**Comment by shyamjvs** [comment]
Gathered some interesting evidence today. Look at how the CPU usage of controller-manager increased ~4x across runs 454 and 455:

![controller-manager-with-quotas](https://user-images.githubusercontent.com/4333971/37280904-77a27346-25ef-11e8-8b1e-522faad3b62b.png)

**Comment by shyamjvs** [comment]
From the [commit diff](https://github.com/kubernetes/kubernetes/compare/3d19fe401...73b2b1cac), it seems most likely that enabling quotas (#60421) caused the mischief. Also, I looked at controller-manager logs from one of the newer runs and find many such lines:

```
E0310 00:16:53.089141       1 resource_quota_controller.go:255] Operation cannot be fulfilled on resourcequotas "e2e-tests-load-30-nodepods-1-vxb5m-quota": the object has been modified; please apply your changes to the latest version and try again
E0310 00:16:53.137643       1 resource_quota_controller.go:255] Operation cannot be fulfilled on resourcequotas "e2e-tests-load-30-nodepods-1-vxb5m-quota": the object has been modified; please apply your changes to the latest version and try again
E0310 00:16:53.181949       1 resource_quota_controller.go:255] Operation cannot be fulfilled on resourcequotas "e2e-tests-load-30-nodepods-1-vxb5m-quota": the object has been modified; please apply your changes to the latest version and try again
E0310 00:16:53.257855       1 resource_quota_controller.go:255] Operation cannot be fulfilled on resourcequotas "e2e-tests-load-30-nodepods-1-vxb5m-quota": the object has been modified; please apply your changes to the latest version and try again
```

**Comment by shyamjvs** [comment]
Found one final piece of evidence to prove that it's indeed a problem caused by resource-quotas. Look at how our scalability job running against HEAD saw a fall back to normal CPU usage (across runs 11485 and 11486), with the [only change](https://github.com/kubernetes/kubernetes/compare/7c9293e1c...40143fd68) being disabling of quotas:

![controller-manager without-quotas](https://user-images.githubusercontent.com/4333971/37281285-d10a3fda-25f0-11e8-9812-e85c173064e7.png)

**Comment by shyamjvs** [comment]
cc @davidopp 
(because I somehow remember you associated with the feature:)

**Comment by gmarek** [comment]
cc @deads2k

**Comment by deads2k** [comment]
> cc @deads2k

@derekwaynecarr point of interest
